### ISSUE TITLE
Princess settings toggle by round number and BV percentage

### ISSUE THREAD

**Issue by SuperStucco** [MegaMek/mekhq#1995]
As of latest (47.9) the Princess AI has settings that apply for the entire game.  Changing settings for different behavior (retreat; suicidal defense; etc.) means kicking the AI player then using Replace Player and specifying different settings.

I would like to streamline this a bit with Princess changing states based on either the current round number, or the current BV percentage.

Add the the Princess settings dialog a second tab or panel with the same slider (and retreat edge/auto-retreat off map) settings as current.  Include two options, checking the current round OR a BV percentage.  Once the selected condition is met in the end phase, the new set of Princess settings is used.

When checking the round number, allow for a straight "On round [number]" OR for "When 2d6 + [base number] equals or exceeds the round number".

The compare BV percentage option is straight forward - check the percentage of remaining BV against the target number.

**Comment by Windchild292** [comment]
@NickAragua isn't this a MegaMek Option?

**Comment by SuperStucco** [comment]
Aurghh... was thinking about changing them from inside MekHQ, but you're right this is probably better served there (as long as its still available from the Briefing room tab as before).

**Comment by Windchild292** [comment]
I believe it is the same dialog, but @NickAragua is the Princess Expert.

**Comment by NickAragua** [comment]
Yeah, this would be something implemented in MegaMek. MekHQ might make use of it dynamically at some point, but the bulk of the work and initial implementation would likely be in MegaMek.
